src/expr.rs

```diff
@@ -221,38 +222,108 @@ impl<'a> FmtVisitor<'a> {
         format!("({})", item_str)
     }
 
+    fn rewrite_binary_op(&mut self,
+                         op: &ast::BinOp,
+                         lhs: &ast::Expr,
+                         rhs: &ast::Expr,
+                         width: usize,
+                         offset: usize)
+        -> String
+    {
+        let operator_str = self.snippet(op.span);
+
+        // 1 = space between lhs expr and operator
+        let mut result = self.rewrite_expr(lhs, width - 1 - operator_str.len(), offset);
+
+        result.push(' ');
+        result.push_str(&operator_str);
+
+        let remaining_width = match result.rfind('\n') {
+            Some(idx) => {
+                let space = config!(max_width) + idx;
+                let used = offset + result.len();
```

Comment:
Nope! But...

The reformatted result should include the offset already, e.g,
```
        foo(bar,
            baz) +
```

`result.len() - idx` will give the length of `            baz) +`, i.e., the spillover including the offset. Is that right?